src/xdebugConnection.ts

```diff
@@ -823,3 +827,33 @@ export class Connection extends DbgpConnection {
         return new EvalResponse(await this._enqueueCommand('eval', undefined, expression), this)
     }
 }
+/**
+ * This class represents a connection to Dbgp proxy server for multi user debugging
+ */
+export class DbgpProxyConnection extends Connection {
+    /** Constructs a new connection that uses the given socket to communicate with XDebug. */
+    private _port: number
+    private _host: string
+    private _ideKey: string
+    /** CLI option inputed as binary boolean. */
+    private _allowMultipleSession: number
+
+    constructor(socket: net.Socket, ideKey: string, port: number, host: string, _allowMultipleSession: boolean) {
+        super(socket)
+        this._port = port
+        this._host = host
+        this._ideKey = ideKey
+        this._allowMultipleSession = _allowMultipleSession ? 1 : 0
+        this.sendProxyInitCommand()
+    }
+
+    private async sendProxyInitCommand() {
+        this._enqueueCommand('proxyinit', `-p ${this._port} -k ${this._ideKey} -m ${this._allowMultipleSession}`)
+    }
+
+    public async sendProxyStopCommand() {
+        const proxySocket = net.createConnection({ port: this._port, host: this._host }, () => {
```

Comment:
I am not familiar with how the proxy works, could you explain why this has to create a new socket and not just go through `this._enqueueCommand`?

Reply:
It appears it does not need to. I didn't write this bit but I've just tested it with `this._enqueueCommand` and it appears to be working equally.